scheduler/scheduler.py

```diff
@@ -2666,24 +2669,34 @@ def _update_lease_callback(self, job_id, worker_id, steps, duration,
                                                         max_steps,
                                                         max_duration))
 
-        # Round the remaining steps to the nearest multiple of scale_factor.
-        scale_factor = self._jobs[job_id].scale_factor
-        remaining_steps = self._get_remaining_steps(job_id)
-        remaining_steps = int(math.ceil(remaining_steps / scale_factor))
+            # Round the remaining steps to the nearest multiple of scale_factor.
```

Comment:
You want to access these variables with the lock held, correct?

Reply:
Yes